### ISSUE TITLE
Run program multiple times with different arguments?

### ISSUE THREAD

**Issue by johnnytchuang** [OpenCppCoverage/OpenCppCoverage#56]
**Before creating an issue, please make sure the answer is not in the [FAQ](https://github.com/OpenCppCoverage/OpenCppCoverage/wiki/FAQ).**

**This template is applied for all bug requests. Remove it for other general question or feature request.**

## Describe your environment
- Version of OpenCppCoverage: 0.9.7.0
- Architecture (x86/ x64): x64
- Windows version: 10

## Describe your problem

Is there any way to run our program multiple times with different arguments without having to run OpenCppCoverage.exe multiple times? See an example below in the "toy example" section.

The hope is that we can save time processing / instrumenting modules while still getting the same result as if we had run OpenCppCoverage.exe multiple times on the same program with different arguments each time, generating a new report each time.

Or, perhaps we are going about this the wrong way? What we are doing is running tests and we would like to know what code each test covers. Each test can be run with `program.exe --test1` or `program.exe --test2`, etc. But running all of our (hundreds of) tests separately would take days, with much of that time (we think) spent on re-instrumenting the modules for each run.

Could you reproduce the issue on a toy example?

We would like the same result as if we had done this:
`OpenCppCoverage.exe --sources \path\to\project -- program.exe --arg1`
`OpenCppCoverage.exe --sources \path\to\project -- program.exe --arg2`

But instead of running OpenCppCoverage.exe multiple times it would be faster (we think) to run it once. Perhaps something like this (just making up syntax):
`OpenCppCoverage.exe --sources \path\to\project -- program.exe --arg1 -- program.exe --arg2`

The benefit of this (we hope) is that it would take less time overall because OpenCppCoverage wouldn't have to process / instrument modules each time. In the above example, there are only two variants, but in reality we need hundreds of different runs (and their corresponding reports).

**Comment by OpenCppCoverage** [comment]
Hello,

Unfortunately, OpenCppCoverage does not fit well to your use case.
Indeed, it needs to know the memory locations of each source code line and so it needs to read full pdb each time.

One way to handle your case is that OpenCppCoverage could provide an API function like ``DumpCoverage(filename)``.
This function could be called from your unit tests and dumps the coverage to a specific file. The problem is that it is not small fixes :(.  

Depending on your needs, you can consider this workaround:
  - Run each test and export the coverage to the Cobertura format (XML)
  - Parse the Cobertura files and list all files with at least one line covered.
  - Create a [config file](https://github.com/OpenCppCoverage/OpenCppCoverage/wiki/Command-line-reference#--config_file-arg) that use only the files where at least one line is covered.
  - Run each test with the associated config file.

Is there any way to reuse existing code coverage information to not recompute everything?
   
I hope it will help,

OpenCppCoverage

**Comment by di-papadopoulos** [comment]
I agree with @johnnytchuang , this would be a really nice feature, to be able to load once the OpenCppCoverage.exe and then run the same application with different arguments. Having more than 15000 test cases means we need to load our application 15000 times. It is a lot of wasty time.

**Comment by VShawn** [comment]
much agree
i'd like to run my test once with different args, instead run many tests.

**Comment by OpenCppCoverage** [comment]
I will investigate if caching some data can help.

**Comment by OpenCppCoverage** [comment]
Thank you for reporting this issue and for contributing to the project.

The project is now being formally retired and the repository will be archived. I am therefore closing all remaining issues, as no further investigation, fixes, or releases are planned.

This administrative closure does not mean that the issue was resolved or that the report was invalid. The discussion will remain available for anyone who wants to address it in a fork.

Thank you for your understanding.
